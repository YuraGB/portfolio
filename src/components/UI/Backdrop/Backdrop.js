/**
 * Backdrop
 *
 * @author Yurii Huriianov <yuhur1985@gmail.com
 * @copyright 2020
 */
import React from 'react';
import PropTypes from 'prop-types';

import classes from './Backdrop.module.css';

/**
 * Backdrop
 *
 * @param props
 * @return {*} component
 */
const Backdrop = (props) => {
    return (
        props.show ?
            <div
                className={classes.Backdrop}
                onClick={props.clicked}
            ></div> :
            null
    )
};

Backdrop.propTypes = {
    show: PropTypes.bool,
    clicked: PropTypes.func
};

export default Backdrop;