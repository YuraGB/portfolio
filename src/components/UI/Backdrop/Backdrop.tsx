/**
 * Backdrop
 *
 * @author Yurii Huriianov <yuhur1985@gmail.com
 * @copyright 2020
 */
import React from 'react';

import classes from './Backdrop.module.css';
import {IBackdropInterface} from '../../../system/types';

/**
 * Backdrop
 *
 * @param props
 * @return {*} component
 */
const Backdrop:React.FC<IBackdropInterface> = ({clicked, show}) => {
    return (
        show ?
            <div
                className={classes.Backdrop}
                onClick={clicked}
            ></div> :
            null
    )
};

export default Backdrop;