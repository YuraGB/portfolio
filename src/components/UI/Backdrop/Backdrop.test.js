/**
 * Backdrop unitest
 *
 * @author Yurii Huriianov <yuhur1985@gmail.com
 * @copyright 2020
 */

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import Backdrop from './Backdrop';

configure({adapter: new Adapter()});

describe('<Backdrop />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Backdrop />);
    });

    it('show backdrop if prop.show present', () => {
        wrapper.setProps({show:true});
        expect(wrapper.getElement()).toEqual(<div className='Backdrop'></div> );
    })
});