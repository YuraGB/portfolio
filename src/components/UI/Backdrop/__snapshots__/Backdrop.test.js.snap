// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> show backdrop if prop.show present 1`] = `ShallowWrapper {}`;
